File: server/internal/shared/dto/pagination.go
package shareddto

import "github.com/Watari995/musclead/internal/pagination"

type PaginationDTO struct {
	CurrentPage  int `json:"current_page"`
	ItemsPerPage int `json:"items_per_page"`
	TotalItems   int `json:"total_items"`
	TotalPages   int `json:"total_pages"`
}

func NewPaginationDTO(paginator pagination.OffsetPaginator) PaginationDTO {
	return PaginationDTO{
		CurrentPage:  paginator.CurrentPage,
		ItemsPerPage: paginator.ItemsPerPage,
		TotalItems:   paginator.TotalItems,
		TotalPages:   paginator.TotalPages,
	}
}

File: mobile/lib/core/data/pagination.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'pagination.freezed.dart';
part 'pagination.g.dart';

@freezed
abstract class PaginationDto with _$PaginationDto {
  const factory PaginationDto({
    @Default(1) int currentPage,
    @Default(0) int itemsPerPage,
    @Default(0) int totalItems,
    @Default(0) int totalPages,
  }) = _PaginationDto;

  factory PaginationDto.fromJson(Map<String, dynamic> json) =>
      _$PaginationDtoFromJson(json);
}
